com.unity.ml-agents/Runtime/Sensor/RayPerceptionSensor.cs

```diff
@@ -4,91 +4,245 @@
 
 namespace MLAgents
 {
-    public class RayPerceptionSensor : ISensor
+    /// <summary>
+    /// Determines which dimensions the sensor will perform the casts in.
+    /// </summary>
+    public enum RayPerceptionCastType
     {
-        public enum CastType
-        {
-            Cast2D,
-            Cast3D,
-        }
+        /// Cast in 2 dimensions, using Physics2D.CircleCast or Physics2D.RayCast.
+        Cast2D,
 
-        float[] m_Observations;
-        int[] m_Shape;
-        string m_Name;
+        /// Cast in 3 dimensions, using Physics.SphereCast or Physics.RayCast.
+        Cast3D,
+    }
+
+    public class RayPerceptionInput
```

Comment:
is there a reason this is a class instead of a struct?

Reply:
I originally intended for an instance of this to be shared between the component and ISensor, and I think that might eventually happen (if I have time).

Reply:
I feel like updating a reference type in one class that updates another feels a bit confusing.  I'd rather just set a value type on the sensor every time the struct changes.  I'm not sure if we have a design philosophy in place to back this up.  It could just be an outdated opinion that I have.